sdk/src/clearingHouseUser.ts

```diff
@@ -237,28 +271,58 @@ export class ClearingHouseUser {
 				const market = this.clearingHouse.getMarketAccount(
 					marketPosition.marketIndex
 				);
-				return pnl.add(
-					calculatePositionPNL(
-						market,
-						marketPosition,
-						withFunding,
-						this.getOracleDataForMarket(market.marketIndex)
-					)
+				let pnl0 = calculatePositionPNL(
+					market,
+					marketPosition,
+					withFunding,
+					this.getOracleDataForMarket(market.marketIndex)
 				);
+
+				if (withWeightMarginCategory !== undefined) {
+					if (pnl0.gt(ZERO)) {
+						pnl0 = pnl0.mul(
+							calculateUnsettledAssetWeight(
+								market,
+								pnl0,
+								withWeightMarginCategory
+							)
+						);
+					}
+				}
+
+				return pnl.add(pnl0);
 			}, ZERO);
 	}
 
 	/**
 	 * calculates unrealized position price pnl
 	 * @returns : Precision QUOTE_PRECISION
 	 */
-	public getUnsettledPNL(marketIndex?: BN): BN {
+	public getUnsettledPNL(
```

Comment:
same comments about the param object, marketPnl name and dividing by weight precision